src/http_server.cc

```diff
@@ -1515,6 +1528,10 @@ HTTPAPIServer::HandleModelReady(
     evhtp_request_t* req, const std::string& model_name,
     const std::string& model_version_str)
 {
+  if (RestrictedAPI(req, "health")) {
```

Comment:
could move to a macro if it fits better

Reply:
If this trend of adding checks to headers continues then we may want to think about a full pre-/post-request process implementation where users can opt-in to the checks they want done.

Reply:
For now I'd like to keep this to bringing parity with grpc restrictions - we could generalize later but I would want to make sure we have a use case in mind. reasonable?

The option itself is pretty generic - so can check any header name - but only allows one check per api type ...

We do other checks like method / model name / etc. -

Reply:
That's reasonable. I meant for this to be more of a comment on what we should consider for the future more than an actual change request.